jetty-util/src/main/java/org/eclipse/jetty/util/thread/ScheduledExecutorScheduler.java

```diff
@@ -50,28 +55,48 @@ public ScheduledExecutorScheduler()
 
     public ScheduledExecutorScheduler(String name, boolean daemon)
     {
-        this(name, daemon, Thread.currentThread().getContextClassLoader());
+        this(name, daemon, null);
     }
 
-    public ScheduledExecutorScheduler(String name, boolean daemon, ClassLoader threadFactoryClassLoader)
+    public ScheduledExecutorScheduler(@Name("name") String name, @Name("daemon") boolean daemon, @Name("threads") int threads)
     {
-        this(name, daemon, threadFactoryClassLoader, null);
+        this(name, daemon, null, null, threads);
     }
 
-    public ScheduledExecutorScheduler(String name, boolean daemon, ClassLoader threadFactoryClassLoader, ThreadGroup threadGroup)
+    public ScheduledExecutorScheduler(String name, boolean daemon, ClassLoader classLoader)
+    {
+        this(name, daemon, classLoader, null);
+    }
+
+    public ScheduledExecutorScheduler(String name, boolean daemon, ClassLoader classLoader, ThreadGroup threadGroup)
+    {
+        this(name, daemon, classLoader, threadGroup, -1);
+    }
+
+    /**
+     * @param name The name of the scheduler threads or null for automatic name
+     * @param daemon True if scheduler threads should be daemon
+     * @param classLoader The classloader to run the threads with or null to use the current thread context classloader
+     * @param threadGroup The threadgroup to use or null for no thread group
+     * @param threads The number of threads to pass to the the core {@link ScheduledThreadPoolExecutor} or -1 for a
+     * heuristic determined number of threads.
+     */
+    public ScheduledExecutorScheduler(@Name("name") String name, @Name("daemon") boolean daemon, @Name("classLoader") ClassLoader classLoader, @Name("threadGroup") ThreadGroup threadGroup, @Name("threads") int threads)
     {
         this.name = name == null ? "Scheduler-" + hashCode() : name;
         this.daemon = daemon;
-        this.classloader = threadFactoryClassLoader == null ? Thread.currentThread().getContextClassLoader() : threadFactoryClassLoader;
+        this.classloader = classLoader == null ? Thread.currentThread().getContextClassLoader() : classLoader;
         this.threadGroup = threadGroup;
+        this.threads = threads;
     }
 
     @Override
     protected void doStart() throws Exception
     {
-        scheduler = new ScheduledThreadPoolExecutor(1, r ->
+        int size = threads > 0 ? threads : Math.max(1, ProcessorUtils.availableProcessors() / 4);
```

Comment:
I don't see any reason to have more than 1 thread by default, like it was previously. in large machines we are going to possibly waste tens of threads.